erizo/src/erizo/rtp/RRGenerationHandler.cpp

```diff
@@ -0,0 +1,251 @@
+#include "rtp/RRGenerationHandler.h"
+#include "./WebRtcConnection.h"
+#include "lib/ClockUtils.h"
+
+namespace erizo {
+
+DEFINE_LOGGER(RRGenerationHandler, "rtp.RRGenerationHandler");
+
+RRGenerationHandler::RRGenerationHandler(WebRtcConnection *connection) :
+    temp_ctx_{nullptr}, enabled_{true} {}
+
+
+void RRGenerationHandler::enable() {
+  enabled_ = true;
+}
+
+void RRGenerationHandler::disable() {
+  enabled_ = false;
+}
+
+bool RRGenerationHandler::isRetransmitOfOldPacket(std::shared_ptr<dataPacket> packet) {
+  RtpHeader *head = reinterpret_cast<RtpHeader*>(packet->data);
+  if (packet->type == VIDEO_PACKET) {
+    if (!rtpSequenceLessThan(head->getSeqNumber(), video_rr_.max_seq) || jitter_video_.jitter == 0) {
+      return false;
+    }
+    int64_t time_diff_ms = static_cast<uint32_t>(packet->received_time_ms) - video_rr_.last_recv_ts;
```

Comment:
I think we're still duplicating lines of code between video and audio cases. Is there a possible to create generic functions? let the code as is if you don't find a quick and easy solution.